components/board/fake/board.go

```diff
@@ -233,39 +242,48 @@ func (b *Board) SetPowerMode(ctx context.Context, mode pb.PowerMode, duration *t
 func (b *Board) StreamTicks(ctx context.Context, interrupts []board.DigitalInterrupt, ch chan board.Tick,
 	extra map[string]interface{},
 ) error {
-	for _, i := range interrupts {
-		name := i.Name()
-		d, ok := b.Digitals[name]
-		if !ok {
-			return fmt.Errorf("could not find digital interrupt: %s", name)
-		}
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-			// Keep going
-		}
-		// Get a random bool for the high tick value.
-		// linter complans about security but we don't care if someone
-		// can predict if the fake interrupts will be high or low.
-		//nolint:gosec
-		randBool := rand.Int()%2 == 0
-		select {
-		case ch <- board.Tick{Name: d.conf.Name, High: randBool, TimestampNanosec: uint64(time.Now().Unix())}:
-		default:
-			// if nothing is listening to the channel just do nothing.
-		}
+	for _, di := range interrupts {
+		b.wg.Add(1)
+		go func(name string) {
+			defer b.wg.Done()
+			d, ok := b.Digitals[name]
+			if !ok {
+				b.logger.Errorw("ticks not streamed, digital interrupt not found", "digital_interrupt", name)
+				return
```

Comment:
If someone attempts to stream ticks from a nonexistent interrupt, we're going to return no errors, as though everything worked. I think we should check if the interrupt exists outside of the goroutine, and return errors when that happens.

...part of the confusion here is that we had an interrupt as an argument to `StreamTicks`, then we passed only the name to the goroutine, and are trying to look up the interrupt by name in here. I wonder if we can just use the interrupt directly, and get rid of line 249 entirely. but I haven't thought that last part through yet, and maybe it's not possible.

Reply:
in this case, since it is a fake board, the interrupt itself is unlikely to have anything mocked out for ticks. but I'll return an error if any of the interrupts are missing.

Reply:
updated